### PR TITLE
Travel Pay / flesh out SMOC introduction page

### PR DESCRIPTION
**Pull request by liztownd**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
Add content to Introduction page; refactor Help Text and update where it was used; add mock data for user + appointment

- _(Which team do you work for, does your team own the maintenance of this component?)_
Travel Pay

- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
- Phased rollout beginning Feb 2025

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[#92353](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92353)
[#10016](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100016)
[#10015](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100015)

- _Link to epic if not included in ticket_
[#83829](https://github.com/department-of-veterans-affairs/va.gov-team/issues/83829)

## Testing done

- _Describe what the old behavior was prior to the change_
The Introduction page for our submit mileage claim form flow was a shell, now it has data (hardcoded) and content.
- _Describe the steps required to verify your changes are working as expected_
Locally: Start the mock api with `yarn mock-api --responses src/applications/travel-pay/services/mocks/index.js`; Build locally with `yarn build:webpack:local --env api="http://localhost:3000"` then serve the build with `WEB_PORT=3001 node src/platform/testing/e2e/test-server.js --buildtype=localhost` (instructions in README).
- _Describe the tests completed and the results_
New Cypress test to test the flow through the form; New unit tests to verify each component including the new Introduction Page component.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    Intro mobile old    |     Intro mobile new    |
| Desktop |   Intro page old      |    Intro page new     |

## What areas of the site does it impact?

Travel Pay directory only

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by seesleestak — DISMISSED** [review]

**Review by seesleestak — DISMISSED** [review]
Working well for me 👍

File: src/applications/travel-pay/components/Breadcrumbs.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default function BreadCrumbs() {
 
   const { apptId } = useParams();
 
-  // TODO: this needs work
+  // TODO: this might need work
```
Comment (kjduensing) [inline src/applications/travel-pay/components/Breadcrumbs.jsx:14]: Let's add (briefly) what work might be needed here to make future engineers' jobs easier.

File: src/applications/travel-pay/containers/SubmitFlowWrapper.jsx [hunk 2]
```diff
@@ -13,14 +13,17 @@ import BreadCrumbs from '../components/Breadcrumbs';
...
-  const [cantFile, setCantFile] = useState(false);
-
+  // TODO: Placeholders until backend integration
+  // API call based on the URL Params, but for now is hard coded
+  const appointment = appointment1;
   // This will actually be handled by the redux action, but for now it lives here
   const [isSubmissionError, setIsSubmissionError] = useState(false);
 
   const [pageIndex, setPageIndex] = useState(0);
+  const [cantFile, setCantFile] = useState(false);
```
Comment (kjduensing) [inline src/applications/travel-pay/containers/SubmitFlowWrapper.jsx:26]: `cantFile` reads a little too close to `canFile` IMO. Plus, having a negative in the name of the variable tends to confuse things (e.g. `notValid`) in most cases. 

What do you think about using `canFile` here and have the logic be based around that?

Comment (kjduensing) [reply src/applications/travel-pay/containers/SubmitFlowWrapper.jsx:26]: Also, is `cantFile` in preparation for something? I don't see it used here, but I know there's more stuff coming for this work.

File: src/applications/travel-pay/tests/components/submit-flow/pages/CantFilePage.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { expect } from 'chai';
+import { render } from '@testing-library/react';
+import { $ } from 'platform/forms-system/src/js/utilities/ui';
+
+import CantFilePage from '../../../../components/submit-flow/pages/CantFilePage';
```
Comment (kjduensing) [inline src/applications/travel-pay/tests/components/submit-flow/pages/CantFilePage.unit.spec.jsx:6]: I suppose I should have caught this earlier, but what about `CannotFilePage` to make it more readable?

File: src/applications/travel-pay/util/dates.js [hunk 4]
```diff
@@ -74,3 +74,12 @@ export function getDateFilters() {
 
   return dateRanges;
 }
+
+export function getDaysLeft(datetimeString) {
+  const apptDate = new Date(datetimeString);
+  const daysSinceAppt = Math.floor(
+    (new Date() - apptDate) / 1000 / 60 / 60 / 24,
+  );
```
Comment (kjduensing) [inline src/applications/travel-pay/util/dates.js:82]: A couple different options using date-fns:

Requires us to validate that it's 30 business days, not calendar days:
`const daysSinceAppt = differenceInBusinessDays(new Date(), apptDate)`

Alternatively:
`const daysSinceAppt = differenceInCalendarDays(new Date(), apptDate)`

or if measuring pure 24 hour periods:
`const daysSinceAppt = differenceInDays(new Date(), apptDate)`

File: src/applications/travel-pay/util/dates.js [hunk 5]
```diff
@@ -74,3 +74,12 @@ export function getDateFilters() {
 
   return dateRanges;
 }
+
+export function getDaysLeft(datetimeString) {
```
Comment (kjduensing) [inline src/applications/travel-pay/util/dates.js:78]: I'd like to see some tests for this since we're adding logic

File: src/applications/travel-pay/containers/SubmitFlowWrapper.jsx (hunk 2 above)
Comment (liztownd) [reply src/applications/travel-pay/containers/SubmitFlowWrapper.jsx:26]: I have never been a fan of the whole "can't file" nomenclature. What about something more along the lines of "unsupported type"; i.e. `isUnsupportedType`, etc.? (And the page can be re-named to "UnsupportedTypePage"?)

File: src/applications/travel-pay/util/dates.js (hunk 4 above)
Comment (liztownd) [reply src/applications/travel-pay/util/dates.js:82]: Is it business days or calendar days though? I thought it was calendar days? (If it is calendar days, then yeah, I can try using the `differenceInCalendarDays`.)

Or if we want to go with 24 hour periods, like if their appt was at 3pm, and they try to file at 9am on the 30th day, should we allow it to go through? 🤔

File: src/applications/travel-pay/util/dates.js (hunk 5 above)
Comment (liztownd) [reply src/applications/travel-pay/util/dates.js:78]: Good call. I'll add some.

File: src/applications/travel-pay/util/dates.js (hunk 4 above)
Comment (kjduensing) [reply src/applications/travel-pay/util/dates.js:82]: That's why I mentioned needing to validate. Either way there's an easier way to do this than manually compute.

File: src/applications/travel-pay/containers/SubmitFlowWrapper.jsx (hunk 2 above)
Comment (kjduensing) [reply src/applications/travel-pay/containers/SubmitFlowWrapper.jsx:26]: Yeah I like that. I'd prefer something more specific, for example: `isUnsupportedClaimType`

File: src/applications/travel-pay/util/dates.js (hunk 4 above)
Comment (liztownd) [reply src/applications/travel-pay/util/dates.js]: For now, unless we hear differently from VTP I went with calendar days.

File: src/applications/travel-pay/tests/components/submit-flow/pages/CantFilePage.unit.spec.jsx (hunk 3 above)
Comment (kjduensing) [reply src/applications/travel-pay/tests/components/submit-flow/pages/CantFilePage.unit.spec.jsx:6]: I recommend `UnsupportedClaimTypePage`

File: src/applications/travel-pay/tests/util/dates.spec.js [hunk 6]
```diff
@@ -205,3 +205,21 @@ describe('getDateFilters', () => {
     expect(dateRanges.length).to.eq(6);
   });
 });
+
+describe('getDaysLeft', () => {
```
Comment (kjduensing) [inline src/applications/travel-pay/tests/util/dates.spec.js:209]: Thanks for adding these - 2 things:

1. Just to complete the edge cases, I'd like to see a test for if a user has exactly 30 days left
2. what thought did you put into timezones with the days left?

Comment (liztownd) [reply src/applications/travel-pay/tests/util/dates.spec.js:209]: 1. ✅ 
2. Exactly no thought. 😄  (Do I need to? Not sure how the `new Date` takes timezones into consideration. If you think it needs to be accounted for I can play with it, but since we're using the Calendar Days vs. Hours check I'm not sure how much difference it might make unless there's the edge case of someone trying to file like right before midnight just before the clock runs out....)

Comment (liztownd) [reply src/applications/travel-pay/tests/util/dates.spec.js:209]: Ok, so I've done a little research. the `new Date` fn doesn't take timezones into account, and I'm passing in the non-timezoned version of the appt date (the `start` datetime string vs. the `localStartTime` value) so I think it should still give the correct days left response.

**Review by kjduensing — DISMISSED** [review]
Looks good! 🍌

**Review by kjduensing — APPROVED** [review]

**Review by ataker — APPROVED** [review]
